ra/ra.go

```diff
@@ -1721,9 +1848,18 @@ func (ra *RegistrationAuthorityImpl) DeactivateAuthorization(ctx context.Context
 	if auth.Status != core.StatusValid && auth.Status != core.StatusPending {
 		return berrors.MalformedError("only valid and pending authorizations can be deactivated")
 	}
-	err := ra.SA.DeactivateAuthorization(ctx, auth.ID)
-	if err != nil {
-		return berrors.InternalServerError(err.Error())
+	if features.Enabled(features.NewAuthorizationSchema) {
+		authzID, err := strconv.ParseInt(auth.ID, 10, 64)
```

Comment:
This assumes that immediately on turning on the NewAuthorizationSchema, all authorizations will be v2-style. Actually, we need to decide which code path to take based on whether auth.v2 is true.